docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,513 @@
+
+The first thing you'll need in order to implement this solution is to install [http-proxy](https://www.npmjs.com/package/http-proxy):
+
+```sh
+$ npm i http-proxy
+```
+
+Then, you'll want to create an endpoint in your application which will serve as a proxy for all the requests directed at Contentful's API.
+
+For this, you'll have to create the file `pages/api/cms/[...cms].js`. This file/directory structure and naming is important because you'll want to receive any requests directed to `<hostname>/api/cms/*`.
+
+This endpoint will create the `proxy` server and, on each request, remove `/api/cms` from the request, rewrite the `host` header to the correct host (`cdn.contentful.com`), redirect the request to `cdn.contentful.com` and set the `Cache-Control` HTTP header of the response to `s-maxage=60`.
+
+```js
+import httpProxy from 'http-proxy';
+
+const proxy = httpProxy.createProxyServer({});
+
+export default (req, res) => {
+    req.url = req.url.replace('/api/cms', '');
+    req.headers.host = 'cdn.contentful.com';
+
+    proxy.web(req, res, {
+        target: {
+            host: 'cdn.contentful.com',
+        },
+    });
+
+    if (req.method === 'GET') {
+        res.setHeader('Cache-Control', 's-maxage=60');
+    }
+};
+```
+
+Finally, you will need to setup the Contentful client to direct it's requests to the endpoint you just configured. Please note that you only want to cache the requests when in a production environment.
+
+```js
+const clientOptions = {
+    space: process.env.CONTENTFUL_SPACE_ID,
+    accessToken: process.env.CONTENTFUL_TOKEN,
+};
+
+if (process.env.NODE_ENV === 'production') {
+    clientOptions.host = process.env.SITE_URL;
+    clientOptions.basePath = '/api/contentful';
+}
+
+const client = createClient(clientOptions);
+```
+
+**NOTES:**
+> ❗️ You have to be able to access to `process.env.NODE_ENV` from the client side as well as the server side, to proxy client side requests to Contentful in production environments.
+
+> ❗️ The `process.env.SITE_URL` variable has to be correctly configured and accessible from both server and client side, otherwise the request to the proxy endpoint will not happen correctly. This may mean that preview url's in merge request will not function correctly, which will happen if the application is started with `process.env.NODE_ENV` set to `production`.
+
+## Custom SEO
+
+There may be cases where you will want to configure custom SEO per page. Unfortunately, Contentful does not provide out-of-the-box SEO support, so you will need to implement your own strategy. The approach we suggest is the following:
+
+- Create a content model for SEO.
+- Add a field `Title` that should be only used to identify the model, e.g. "Homepage SEO".
+- Add a SEO field (not-rich long text) to the model. Here, the SEO related tags (title, meta, etc.) will be defined in a _json_ format like so:
```

Comment:
```suggestion
- Add a SEO field (json type) to the model. Here, the SEO related tags (title, meta, etc.) will be defined in a _json_ format like so:
```

Reply:
I understand the motivation behind the suggesting, and "not-rich long text" should be changed. But the reasoning behind this was to indicate the type of input (Contentful-wise) you have to create.

Perhaps just leaving as "long text"?

Reply:
![imagem](https://user-images.githubusercontent.com/11398440/79763980-ae9bf200-831c-11ea-8fb1-1fdff3c7ec28.png)

Contentful-wise is `JSON object` :+1:

Reply:
🤦 didn't see it